### ISSUE TITLE
Upgrading from 2.10.4 to 2.11.0 causes 502 for API endpoints

### ISSUE THREAD

**Issue by wdhdev** [NginxProxyManager/nginx-proxy-manager#3473]
**Checklist**
- Have you pulled and found the error with `jc21/nginx-proxy-manager:latest` docker image?
  - Yes
- Are you sure you're not using someone else's docker image?
  - Yes
- Have you searched for similar issues (both open and closed)?
  - Yes

**Describe the bug**

When I run `docker compose pull`, then `docker compose up -d` on my NPM instance and then go to the login page and try to login, it responds with a 502 error. The login page loads and all proxy hosts seem to be online, it's just the API endpoints.

I viewed the logs using `docker compose logs` and it stops at:
```
[1/19/2024] [5:15:29 AM] [Global   ] › ℹ  info      Using Sqlite: /data/database.sqlite
[1/19/2024] [5:15:29 AM] [Migrate  ] › ℹ  info      Current database version: none
```

I tried booting up a brand new instance of 2.11.0 and it works fine.

**Nginx Proxy Manager Version**

Upgrading from 2.10.4 to 2.11.0

**To Reproduce**
1. Have an existing NPM instance on 2.10.4 with some proxy hosts
2. Attempt to upgrade it to 2.11.0
3. See 502 Bad Gateway error on login screen

**Expected behavior**

I expected to be able to login and for the app to load properly.

**Screenshots**

![image](https://github.com/NginxProxyManager/nginx-proxy-manager/assets/87287585/cc932455-94c2-4ba8-aca1-bceca51464c7)
![image](https://github.com/NginxProxyManager/nginx-proxy-manager/assets/87287585/12c76f76-c251-4f4e-bd69-589a5384095d)

**Operating System**

Ubuntu 22 LTS 64-bit

**Additional context**

**Comment by DieMetRik** [comment]
i have the same issue. Can't access to Web
Proxmox -> Debian 12 (LXC) -> docker 24.0.7

**Comment by wdhdev** [comment]
I forgot to add, I'm using a Proxmox VM as well.

**Comment by jacobstein123** [comment]
Same here, but not on Proxmox. I'm on Ubuntu Server using the recommended docker-compose.yml in the docs.

**Comment by fixi87** [comment]
Same for me after upgrading on portainer/docker using the recommended yaml with minimal settings.

**Comment by skipperkm** [comment]
hi,

i have the Same Problem. Update to 2.11.0 an can Not login. I think we Need a Version 2.11.1 😂

**Comment by pr0xyzer** [comment]
Same here

**Comment by bazziebaz** [comment]
same here, downgrade to 2.10.4 fixes the problem

**Comment by przybylas** [comment]
thx, downgrade to 2.10.4 fixes the problem

**Comment by wdhdev** [comment]
> same here, downgrade to 2.10.4 fixes the problem

That's what I've done for now, but I would like to see a solution or fix for this still.

**Comment by jpdsc** [comment]
```
[1/19/2024] [5:15:29 AM] [Global   ] › ℹ  info      Using Sqlite: /data/database.sqlite
[1/19/2024] [5:15:29 AM] [Migrate  ] › ℹ  info      Current database version: none
```

When this appear and you wait around 3-4 min it will show the error why it cannot proceed. I would advise to wait.

**Comment by lumper5** [comment]
Rolled back.  Running in LXC container using https://tteck.github.io/Proxmox/ . Upgrading to 2.11.0 produces "bad gateway". Only solution thus far: roll back to recent snapshot or backup prior to upgrade.

**Comment by JBlond** [comment]
Using mariadb the migration of the database hangs.

**Comment by eskiiom** [comment]
Hello,
same issue here on proxmox/docker 
some issue with the backend engine apparently

`  #  node[448]: std::unique_ptr<long unsigned int> node::WorkerThreadsTaskRunner::DelayedTaskScheduler::Start() at ../src/node_platform.cc:68
  #  Assertion failed: (0) == (uv_thread_create(t.get(), start_thread, this))
----- Native stack trace -----
 1: 0xca5430 node::Abort() [node]
 2: 0xca54b0 node::errors::SetPrepareStackTraceCallback(v8::FunctionCallbackInfo<v8::Value> const&) [node]
 3: 0xd24789 node::WorkerThreadsTaskRunner::WorkerThreadsTaskRunner(int) [node]
 4: 0xd248ac node::NodePlatform::NodePlatform(int, v8::TracingController*, v8::PageAllocator*) [node]
 5: 0xc5e5a3  [node]
 6: 0xc5ef34 node::Start(int, char**) [node]
 7: 0x7f7dcf6691ca  [/lib/x86_64-linux-gnu/libc.so.6]
 8: 0x7f7dcf669285 __libc_start_main [/lib/x86_64-linux-gnu/libc.so.6]
 9: 0xbb8d7e _start [node]
./run: line 21:   448 Aborted                 (core dumped) s6-setuidgid "$PUID:$PGID" bash -c "export HOME=$NPMHOME;node --abort_on_uncaught_exception --max_old_space_size=250 index.js"
❯ Starting backend ...
  #  node[452]: std::unique_ptr<long unsigned int> node::WorkerThreadsTaskRunner::DelayedTaskScheduler::Start() at ../src/node_platform.cc:68
  #  Assertion failed: (0) == (uv_thread_create(t.get(), start_thread, this))
----- Native stack trace -----
 1: 0xca5430 node::Abort() [node]
 2: 0xca54b0 node::errors::SetPrepareStackTraceCallback(v8::FunctionCallbackInfo<v8::Value> const&) [node]
 3: 0xd24789 node::WorkerThreadsTaskRunner::WorkerThreadsTaskRunner(int) [node]
 4: 0xd248ac node::NodePlatform::NodePlatform(int, v8::TracingController*, v8::PageAllocator*) [node]
 5: 0xc5e5a3  [node]
 6: 0xc5ef34 node::Start(int, char**) [node]
 7: 0x7f98d8f1f1ca  [/lib/x86_64-linux-gnu/libc.so.6]
 8: 0x7f98d8f1f285 __libc_start_main [/lib/x86_64-linux-gnu/libc.so.6]
 9: 0xbb8d7e _start [node]
./run: line 21:   452 Aborted                 (core dumped) s6-setuidgid "$PUID:$PGID" bash -c "export HOME=$NPMHOME;node --abort_on_uncaught_exception --max_old_space_size=250 index.js"
❯ Starting backend ...
  #  node[456]: std::unique_ptr<long unsigned int> node::WorkerThreadsTaskRunner::DelayedTaskScheduler::Start() at ../src/node_platform.cc:68
  #  Assertion failed: (0) == (uv_thread_create(t.get(), start_thread, this))
----- Native stack trace -----
 1: 0xca5430 node::Abort() [node]
 2: 0xca54b0 node::errors::SetPrepareStackTraceCallback(v8::FunctionCallbackInfo<v8::Value> const&) [node]
 3: 0xd24789 node::WorkerThreadsTaskRunner::WorkerThreadsTaskRunner(int) [node]
 4: 0xd248ac node::NodePlatform::NodePlatform(int, v8::TracingController*, v8::PageAllocator*) [node]
 5: 0xc5e5a3  [node]
 6: 0xc5ef34 node::Start(int, char**) [node]
 7: 0x7f43ada6c1ca  [/lib/x86_64-linux-gnu/libc.so.6]
 8: 0x7f43ada6c285 __libc_start_main [/lib/x86_64-linux-gnu/libc.so.6]
 9: 0xbb8d7e _start [node]
./run: line 21:   456 Aborted                 (core dumped) s6-setuidgid "$PUID:$PGID" bash -c "export HOME=$NPMHOME;node --abort_on_uncaught_exception --max_old_space_size=250 index.js"
❯ Starting backend ...
  #  node[460]: std::unique_ptr<long unsigned int> node::WorkerThreadsTaskRunner::DelayedTaskScheduler::Start() at ../src/node_platform.cc:68
  #  Assertion failed: (0) == (uv_thread_create(t.get(), start_thread, this))
----- Native stack trace -----
 1: 0xca5430 node::Abort() [node]
 2: 0xca54b0 node::errors::SetPrepareStackTraceCallback(v8::FunctionCallbackInfo<v8::Value> const&) [node]
 3: 0xd24789 node::WorkerThreadsTaskRunner::WorkerThreadsTaskRunner(int) [node]
 4: 0xd248ac node::NodePlatform::NodePlatform(int, v8::TracingController*, v8::PageAllocator*) [node]
 5: 0xc5e5a3  [node]
 6: 0xc5ef34 node::Start(int, char**) [node]
 7: 0x7fcdc0cd21ca  [/lib/x86_64-linux-gnu/libc.so.6]
 8: 0x7fcdc0cd2285 __libc_start_main [/lib/x86_64-linux-gnu/libc.so.6]
 9: 0xbb8d7e _start [node]
./run: line 21:   460 Aborted                 (core dumped) s6-setuidgid "$PUID:$PGID" bash -c "export HOME=$NPMHOME;node --abort_on_uncaught_exception --max_old_space_size=250 index.js"
❯ Starting backend ...
  #  node[464]: std::unique_ptr<long unsigned int> node::WorkerThreadsTaskRunner::DelayedTaskScheduler::Start() at ../src/node_platform.cc:68
  #  Assertion failed: (0) == (uv_thread_create(t.get(), start_thread, this))
----- Native stack trace -----
 1: 0xca5430 node::Abort() [node]
 2: 0xca54b0 node::errors::SetPrepareStackTraceCallback(v8::FunctionCallbackInfo<v8::Value> const&) [node]
 3: 0xd24789 node::WorkerThreadsTaskRunner::WorkerThreadsTaskRunner(int) [node]
 4: 0xd248ac node::NodePlatform::NodePlatform(int, v8::TracingController*, v8::PageAllocator*) [node]
 5: 0xc5e5a3  [node]
 6: 0xc5ef34 node::Start(int, char**) [node]
 7: 0x7f2fd16471ca  [/lib/x86_64-linux-gnu/libc.so.6]
 8: 0x7f2fd1647285 __libc_start_main [/lib/x86_64-linux-gnu/libc.so.6]
 9: 0xbb8d7e _start [node]
./run: line 21:   464 Aborted                 (core dumped) s6-setuidgid "$PUID:$PGID" bash -c "export HOME=$NPMHOME;node --abort_on_uncaught_exception --max_old_space_size=250 index.js"
❯ Starting backend ...
  #  node[468]: std::unique_ptr<long unsigned int> node::WorkerThreadsTaskRunner::DelayedTaskScheduler::Start() at ../src/node_platform.cc:68
  #  Assertion failed: (0) == (uv_thread_create(t.get(), start_thread, this))
----- Native stack trace -----
 1: 0xca5430 node::Abort() [node]
 2: 0xca54b0 node::errors::SetPrepareStackTraceCallback(v8::FunctionCallbackInfo<v8::Value> const&) [node]
 3: 0xd24789 node::WorkerThreadsTaskRunner::WorkerThreadsTaskRunner(int) [node]
 4: 0xd248ac node::NodePlatform::NodePlatform(int, v8::TracingController*, v8::PageAllocator*) [node]
 5: 0xc5e5a3  [node]
 6: 0xc5ef34 node::Start(int, char**) [node]
 7: 0x7f0dbd0cb1ca  [/lib/x86_64-linux-gnu/libc.so.6]
 8: 0x7f0dbd0cb285 __libc_start_main [/lib/x86_64-linux-gnu/libc.so.6]
 9: 0xbb8d7e _start [node]
./run: line 21:   468 Aborted                 (core dumped) s6-setuidgid "$PUID:$PGID" bash -c "export HOME=$NPMHOME;node --abort_on_uncaught_exception --max_old_space_size=250 index.js"`

**Comment by labodj** [comment]
Same here, using external mariadb (latest version with auto upgrade on) hangs, it's been like this for 1 hour, so no errors, no progress on db migration

![Screenshot_20240119-093034.png](https://github.com/NginxProxyManager/nginx-proxy-manager/assets/2527836/96731ac4-d73c-4d40-8d63-3251abb45512)

```yaml
version: "3.9"

secrets:
  mariadb_labo_pwd:
    file: /home/labo/docker/.secrets/mariadb_labo_pwd.txt

services:
  nginx_proxy_manager:
    container_name: NginxProxyManager
    image: jc21/nginx-proxy-manager:latest
    restart: unless-stopped
    networks:
      - bridge_labo
    ports:
      # Public HTTP Port:
      - 80:80
      # Public HTTPS Port:
      - 443:443
      - 443:443/udp
      # Admin Web Port:
      - 81:81
    environment:
      PUID: 1000
      PGID: 1000
      TZ: Europe/Rome
      # These are the settings to access your db
      DB_MYSQL_HOST: mariadb
      DB_MYSQL_PORT: 3306
      DB_MYSQL_USER: redacted
      DB_MYSQL_PASSWORD__FILE: /run/secrets/mariadb_labo_pwd
      DB_MYSQL_NAME: npm
      DISABLE_IPV6: "true"
    volumes:
      - /home/labo/docker/npm/data:/data
      - /home/labo/docker/npm/letsencrypt:/etc/letsencrypt
    secrets:
      - mariadb_labo_pwd
    logging:
      options:
        max-size: 10m
        max-file: "5"
    healthcheck:
      test: ["CMD", "/bin/check-health"]
      interval: 10s
      timeout: 3s

networks:
  bridge_labo:
    external: true
```

**Comment by th12289** [comment]
I have the same issue von VMware, Ubuntu 22.04

version: '3'
services:
  app:
    image: 'jc21/nginx-proxy-manager:latest'
    restart: unless-stopped
    ports:
      - '80:80'
      - '81:81'
      - '443:443'
    volumes:
      - npm_data:/data
      - npm_letsencrypt:/etc/letsencrypt

volumes:
 npm_data:
 npm_letsencrypt:

**Comment by battaglin** [comment]
> thx, downgrade to 2.10.4 fixes the problem

Same problem here.. 
how do you downgrade tho ? 

Thanks

*using docker compose*

**Comment by bazziebaz** [comment]
> Same problem here.. how do you downgrade tho ?

In my case: i changed 'latest' to '2.10.4' in my docker compose file. Restarted the thing.
``version: '3'
services:
  app:
    image: 'jc21/nginx-proxy-manager:2.10.4'
    restart: unless-stopped
    ports:
      - 80:80
      - 40000:81
      - 443:443
    volumes:
      - data:/data
      - letsencrypt:/etc/letsencrypt
volumes:
  data:
  letsencrypt:``

**Comment by tml89** [comment]
same here: ttek Proxmox Script

**Comment by journiyoon** [comment]
When logging in through NPM GUI, error 502 occurred.
After checking fallback_error.log, http://127.0.0.1:3000 connection refused occurred.
I eventually discovered that the backend server wasn't running, but there was nothing I could do.
It was best to change the image version from latest to 2.10.4.

**Comment by lumper5** [comment]
> > Same problem here.. how do you downgrade tho ?
> 
In my case, restore from snapshot I took in Proxmox before upgrading.  I never upgrade - anything - w/o at least a snapshot or full backup. That saved me this time, as it has dozens of times previously.

**Comment by Jamyz** [comment]
I have same issue.
But is only when i configured somes host and restart the container.

**Comment by jicho** [comment]
I'm a bit unsure if I should try the upgrade after reading all the bad experiences above...

When I run the following compose without from scratch NPM starts without any issues on my Synology NAS with latest DSM version and (Synology) Docker:

```
version: "3.8"
services:
  nginx-proxy-manager:
    container_name: nginx-proxy-manager-test
    ports:
      - 20443:443
      - 20080:80
      - 20081:81
    environment:
      - TZ=Europe/Amsterdam
    volumes:
      - /volume1/docker/npm-test/data:/data
      - /volume1/docker/npm-test/le:/etc/letsencrypt
    restart: on-failure:2
    image: jc21/nginx-proxy-manager:2.11.0
networks: {}
```

I can even go to `: 20081` and login with the auto-created admin account.

I might test my luck this weekend with an upgrade...

**Comment by MiRen76** [comment]
Same issue.

**Comment by CorneliusCornbread** [comment]
I'm getting insanely frustrated with how consistently this project breaks between updates, I've had to recreate my proxy setup on several occasions because of breakages between versions. I'm getting _really_ fed up, this is probably the 4th time in about a year of using this I've had to downgrade to solve a bug.

**Comment by bogchop** [comment]
Some problem here on unraid platform.

**Comment by Gunsi1973** [comment]
same here, Proxmox. Downgrade did work. Using SQLite.

**Comment by mweth** [comment]
same

**Comment by JrZavaschi** [comment]
Same here, its bad.

**Comment by cositech** [comment]
same here

**Comment by Momentum6890** [comment]
The same problem here. I have also tried a new installation. The initial login works, but when you log out and try to log in again, all you get is "Bad Gateway".

**Comment by ToxicToxster** [comment]
I posted a bug fix for implementation in #3475

**Comment by jc21** [comment]
Closing this issue in favor of #3475 where I'll be continuing the discussion.
